jx preview fails for repo in bitbucket server

### Summary 

jx preview fails for quickstart repo in bitbucket server

### Steps to reproduce the behavior

Install Jenkins X as Server on EKS, use Bitbucket Server as repository for quickstart

### Expected behavior

jx preview succeeds

### Actual behavior

jx preview fails
On the surface it looks like it is using the term "overview" instead of the actual repository name. Is it a possible but for bitbucketserver integrations?

I have provided the term ```<org>``` instead of the actual BitBucket organisation

```
+ jx preview --app demo-app --dir ../..
Creating a preview
WARNING: issue getting pull request <org>, overview, 3: Status: 404 , Body: {"errors":[{"context":null,"message":"Repository <org>/overview does not exist.","exceptionName":"com.atlassian.bitbucket.repository.NoSuchRepositoryException"}]}
WARNING: Unable to get commits: Status: 404 , Body: {"errors":[{"context":null,"message":"Repository <org>/overview does not exist.","exceptionName":"com.atlassian.bitbucket.repository.NoSuchRepositoryException"}]}
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x98 pc=0x370dbec]

goroutine 1 [running]:
github.com/jenkins-x/jx/pkg/cmd/preview.(*PreviewOptions).Run(0xc000978000, 0x4cacd01, 0xa)
	/workspace/source/pkg/cmd/preview/preview.go:321 +0x392c
github.com/jenkins-x/jx/pkg/cmd/preview.NewCmdPreview.func1(0xc000967680, 0xc0008b2c00, 0x0, 0x4)
	/workspace/source/pkg/cmd/preview/preview.go:134 +0xaa
github.com/spf13/cobra.(*Command).execute(0xc000967680, 0xc0008b2bc0, 0x4, 0x4, 0xc000967680, 0xc0008b2bc0)
	/workspace/go/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:830 +0x2ae
github.com/spf13/cobra.(*Command).ExecuteC(0xc000655400, 0xc0007c2040, 0x59d65c0, 0xc0000bc000)
	/workspace/go/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:914 +0x2fc
github.com/spf13/cobra.(*Command).Execute(...)
	/workspace/go/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:864
github.com/jenkins-x/jx/cmd/jx/app.Run(0x0, 0x0, 0x0, 0xc0008b1f88, 0xc0000a2058)
	/workspace/source/cmd/jx/app/jx.go:19 +0x185
main.main()
	/workspace/source/cmd/jx/jx.go:11 +0x32
````

I believe GitInfo.Name is being populated incorrectly. See https://github.com/jenkins-x/jx/blob/3032216788e7fd1a720a5eab56e52cad63fd0dba/pkg/cmd/preview/preview.go#L291

### Jx version

The output of `jx version` is:

```
jx                 2.0.1100
jenkins x platform 2.0.1813
Kubernetes cluster v1.14.9-eks-c0eccc
kubectl            v1.15.3
helm client        Client: v2.14.3+g0e7f3b6
git                2.22.0
Operating System   Mac OS X 10.14.6 build 18G95
```

### Jenkins type

<!--
Select which installation type are you using. 
-->
- [ ] Serverless Jenkins X Pipelines (Tekton + Prow)
- [ X ] Classic Jenkins

### Kubernetes cluster

EKS - Created using eksctl

### Operating system / Environment

<!--
In which environment are you running the jx CLI?
-->
On Jenkins Server

--- a/pkg/gits/git_url.go
+++ b/pkg/gits/git_url.go
@@ -200,6 +200,12 @@ func parsePath(path string, info *GitRepository, requireRepo bool) (*GitReposito
 	// This is necessary for Bitbucket Server in some cases.
 	trimPath := strings.TrimPrefix(path, "/scm")
 
+	// This is necessary for Bitbucket Server, EG: /projects/ORG/repos/NAME/pull-requests/1/overview
+	reOverview := regexp.MustCompile("/pull-requests/[0-9]+/overview$")
+	if reOverview.MatchString(trimPath) {
+		trimPath = strings.TrimSuffix(trimPath, "/overview")
+	}
+
 	// This is necessary for Bitbucket Server in other cases
 	trimPath = strings.Replace(trimPath, "/projects/", "/", 1)
 	trimPath = strings.Replace(trimPath, "/repos/", "/", 1)

--- a/pkg/gits/git_url_test.go
+++ b/pkg/gits/git_url_test.go
@@ -110,10 +110,16 @@ func TestParseGitURL(t *testing.T) {
 			"git@github.com:bar/foo", "github.com", "bar", "foo",
 		},
 		{
+			"git@github.com:bar/overview", "github.com", "bar", "overview",
+		},
+		{
 			"git@gitlab.com:bar/subgroup/foo", "gitlab.com", "bar", "foo",
 		},
 		{
 			"https://gitlab.com/bar/subgroup/foo", "gitlab.com", "bar", "foo",
+		},
+		{
+			"https://gitlab.com/bar/subgroup/overview", "gitlab.com", "bar", "overview",
 		},
 		{
 			"bar/foo", "github.com", "bar", "foo",
@@ -123,6 +129,9 @@ func TestParseGitURL(t *testing.T) {
 		},
 		{
 			"https://bitbucketserver.com/projects/myproject/repos/foo/pull-requests/1", "bitbucketserver.com", "myproject", "foo",
+		},
+		{
+			"https://bitbucketserver.com/projects/myproject/repos/foo/pull-requests/1/overview", "bitbucketserver.com", "myproject", "foo",
 		},
 	}
 	for _, data := range testCases {

